### DISCUSSION TITLE
Order of sending/receving message with `tokio::mpsc`

### DISCUSSION THREAD

**Discussion by vnghia** [tokio-rs/tokio#6830] [Q&A] [answered]
I would like to ask if there are any guarantee about the order of sent and received message (FIFO) with `tokio::mpsc` in the case of `spsc`. Will message that is sent first always be received first (given that the capacity is bigger than 1) ?

Thank you very much!

**Comment by Darksonn** [comment, marked as answer]
Yes, if msg1 is sent before msg2, then msg1 is received before msg2. This holds with any number of senders.

**Reply by vnghia** [reply]
Thank you. Is there any doc state that or it is just an implementation detail ?

**Reply by Darksonn** [reply]
The documentation says that it's a queue. It would not be a queue if it behaved any other way. It is guaranteed to not change in the future.
